ncm-afsclt/src/main/perl/afsclt.pm

```diff
@@ -2,997 +2,306 @@
-#*** so we put that directory first in the path.
-if ($OSname=~ /Solaris/) {$ENV{'PATH'}="/usr/afsws/bin:$ENV{'PATH'}"}
+my $mypath = '/software/components/afsclt';
 
-sub authconfig_OK {
-  my ($self,$authconfig)=@_;
-
-  # This should be more elaborate 
-
-  unless (-x $authconfig) {
-    $self->error ("$authconfig not found");
-    return 0;
-  }
-  return 1;
-}
-
-##########################################################################
 sub Configure {
-##########################################################################
-#***    In Solaris authconfig is not used so Configure_Cell is
-#*** not needed. The firewall is not enabled so we don't run 
-#*** Configure_firewall. In addition, the afs configuration files are
-#*** not updated from the CDB so we don't use Configure_Config.
-   my ($self,$config)=@_;
-   if ($OSname=~ /Linux/) {
-      $self->Configure_PAM($config);
-      $self->Configure_Cell($config);
-      $self->Configure_firewall($config);
-      $self->Configure_Config($config);
-  }
-  $self->Configure_Cache($config);
-  $self->Configure_CellServDB($config);
+    my ( $self, $config ) = @_;
+    $self->Configure_Cell($config);
+    $self->Configure_TheseCells($config);
+    $self->Configure_Config($config);
+    $self->Configure_Cache($config);
+    $self->Configure_CellServDB($config);
+    $self->Configure_Afsd_Args($config);
 }
 
-##########################################################################
 sub Unconfigure {
-##########################################################################
-  my ($self,$config)=@_;
-  $self->Unconfigure_Cache($config);
-  if ($OSname=~ /Linux/) {
-      $self->Unconfigure_Config($config);
-      $self->Unconfigure_firewall($config);
-      $self->Unconfigure_Cell($config);
-  }
+    my ( $self, $config ) = @_;
 }
 
-##########################################################################
 sub Configure_Cell {
-##########################################################################
-  my ($self,$config)=@_;
-  my $thiscell = "/usr/vice/etc/ThisCell";
-  my $current_cell;
-  unless ($config->elementExists($mypath."/thiscell")) {
-    $self->error("cannot get $mypath/thiscell");
-    return;
-  }
-
-  my $afscell=$config->getValue($mypath."/thiscell");
-
-  if (open(THISCELL, "<", $thiscell)) {
-    $current_cell = <THISCELL>;
-    close(THISCELL);
-    chomp($current_cell);
-  } else {
-    $self->warn("cannot read $thiscell: $!");
-  }
-
-  if ($current_cell && $current_cell eq $afscell) {
-    $self->info("actual cell '$current_cell' in $thiscell is OK");
-  } elsif (! $NoAction) {
-    if ($current_cell) {
-      $self->info("replacing '$current_cell' with '$afscell' in $thiscell");
-    }
-    unless (open(THISCELL, ">", $thiscell)) {
-      $self->error("cannot write $thiscell: $!");
-      return;
+    my ( $self, $config ) = @_;
+    unless ( $config->elementExists("$mypath/thiscell") ) {
+        $self->error("Cannot get $mypath/thiscell");
```

Comment:
make the message clearer that this is a problem in the profile, not woth the component or the system itself (as this is mandatory in the schema)